client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -1203,8 +1203,14 @@ const Editor = (props: EditorProps): JSX.Element => {
     });
   }
 
-  const { theme } = props;
-  const editorTheme = theme === Themes.Night ? 'vs-dark-custom' : 'vs-custom';
+  const { isSignedIn, theme } = props;
+  const preferDarkScheme = window.matchMedia(
+    '(prefers-color-scheme: dark)'
+  ).matches;
+  const editorTheme =
+    theme === Themes.Night || (preferDarkScheme && !isSignedIn)
+      ? 'vs-dark-custom'
+      : 'vs-custom';
```

Comment:
Let's increase condition readability a bit:
```suggestion
  const isDarkTheme = theme === Themes.Night || (preferDarkScheme && !isSignedIn);
  const editorTheme = isDarkTheme ? 'vs-dark-custom' : 'vs-custom';
```